packages/connector/src/services/accounts/services/accounts.ts

```diff
@@ -1,117 +1,723 @@
+          throw new InsufficientBalanceError()
+        // TODO handle other errors
+      }
+    }
+  }
+
+  public async deposit(accountId: string, amount: bigint): Promise<void> {
+    const { assetCode, assetScale, balanceId } = await Account.query()
+      .findById(accountId)
+      .select('assetCode', 'assetScale', 'balanceId')
+      .throwIfNotFound()
+    await this.createTransfers([
+      {
+        sourceBalanceId: toSettlementId(assetCode, assetScale),
+        destinationBalanceId: uuidToBigInt(balanceId),
+        amount
+      }
+    ])
+  }
+
+  public async withdraw(accountId: string, amount: bigint): Promise<void> {
+    const { assetCode, assetScale, balanceId } = await Account.query()
+      .findById(accountId)
+      .select('assetCode', 'assetScale', 'balanceId')
+      .throwIfNotFound()
+
+    await this.createTransfers([
+      {
+        sourceBalanceId: uuidToBigInt(balanceId),
+        destinationBalanceId: toSettlementId(assetCode, assetScale),
+        amount
+      }
+    ])
+  }
+
+  public async getAccountByToken(token: string): Promise<IlpAccount | null> {
+    const account = await Account.query()
+      .withGraphJoined('incomingTokens')
+      .where('incomingTokens.token', token)
+      .first()
+    return account ? toIlpAccount(account) : null
+  }
+
+  public async getAccountByDestinationAddress(
+    destinationAddress: string
+  ): Promise<IlpAccount | null> {
+    try {
+      const account = await Account.query()
+        // new RegExp('^' + staticIlpAddress + '($|\\.)'))
+        .where(
+          raw('?', [destinationAddress]),
+          'like',
+          raw("?? || '%'", ['staticIlpAddress'])
+        )
+        .andWhere((builder) => {
+          builder
+            .where(
+              raw('length(??)', ['staticIlpAddress']),
+              destinationAddress.length
+            )
+            .orWhere(
+              raw('substring(?, length(??)+1, 1)', [
+                destinationAddress,
+                'staticIlpAddress'
+              ]),
+              '.'
+            )
+        })
+        .first()
+      if (account) {
+        return toIlpAccount(account)
+      }
+      const idx = this.config.peerAddresses.findIndex((peer: Peer) =>
+        new RegExp('^' + peer.ilpAddress + '($|\\.)').test(destinationAddress)
+      )
+      if (idx !== -1) {
+        const account = await Account.query().findById(
+          this.config.peerAddresses[idx].accountId
+        )
+        if (account) {
+          return toIlpAccount(account)
+        }
+      }
+      const found = destinationAddress.match(
+        new RegExp(
+          '(?<=^' + this.config.ilpAddress + '\\.)([a-zA-Z0-9_~-]+)(?=($|[.]))'
+        )
+      )
+      if (found) {
+        const account = await Account.query().findById(found[0])
+        if (account) {
+          return toIlpAccount(account)
+        }
+      }
+      return null
+    } catch {
+      return null
```

Comment:
:thinking: Probably just anything `Account.query()` might throw:
https://vincit.github.io/objection.js/recipes/error-handling.html#error-handling
(Not `NotFoundError` since I'm not doing any `.throwIfNotFound` here)
Would it be better to catch, log, and return null or not catch at all?

Reply:
I'm leaning towards not catching at all -- that way the caller can decide whether or not to report the error.